pyoptsparse/pyIPOPT/pyIPOPT.py

```diff
@@ -217,19 +217,25 @@ def __call__(
             # Define the 4 call back functions that ipopt needs:
             def eval_f(x, user_data=None):
                 fobj, fail = self._masterFunc(x, ["fobj"])
-                if fail == 2:
+                if fail == 1:
+                    fobj = np.array(np.NaN)
+                elif fail == 2:
```

Comment:
can we have the parameter fail different from 1 and 2? Should we catch this case and raise an error? Because if this happens then it can raise a bug difficult to detect :-)